fix(hvac): Add option for demand controlled ventilation

chriswmackey opened the pull request:
I am also adding an option to include a humidifier, which is needed for all cases where there's a humidification setpoint.

This is linked to this change:
https://github.com/ladybug-tools/honeybee-energy/pull/669

FYI @mingbopeng . This is technically a breaking change and you'll see that I added a method to the updater module to upgrade old model JSONs. However, this should hopefully not require much change on your end except exposing the new properties. And I implemented the changes in the Python libraries in such a manner that older Honeybee JSONs can still be imported by the latest libraries.

MingboPeng commented:
Hi @chriswmackey,  these changes would cause the issue on C# side when reading json files. 
We use this type value to find the correct class type and then use its de-serializer to read the json.
We will need to have the class names match the type value.
![image](https://user-images.githubusercontent.com/9031066/117170392-fb236580-ad97-11eb-8923-51b54cbd9858.png)

chriswmackey commented:
Thanks @MingboPeng . I am going to revise this PR one more time before merging. It seems my understanding of OpenStudio HVAC was incorrect so the DOAS systems will now be the only one with the class name changed to "Abridged".

chriswmackey commented:
This is the final state of the changes that I was imaging @MingboPeng . I addressed your comments and feel free to do a final review before I merge in an hour or two.

chriswmackey commented:
Actually, I lied. I realized that I want to change the default values of heat recovery and economizer type to be more explicit since I can see now that the standards gem always includes no heat recovery and no economizer by default. I think it's better and easier to make this change now rather than later.

MingboPeng commented:
Thanks, it looks good to me, and thanks for moving DaylightingControl around.
